core/variant.cpp

```diff
@@ -2259,8 +2259,8 @@ Variant::Variant(const RefPtr &p_resource) {
 
 	type = OBJECT;
 	memnew_placement(_data._mem, ObjData);
-	REF ref = p_resource;
-	_get_obj().obj = ref.ptr();
+	REF *ref = reinterpret_cast<REF *>(p_resource.get_data());
```

Comment:
It's simply to have less calls to reference/unreference, which use atomic operations and get propagated to the script system. We just need to get the raw pointer and it can be done very cheaply.